test/e2e/aks.go

```diff
@@ -348,3 +354,119 @@ func GetLatestStableAKSKubernetesVersion(ctx context.Context, subscriptionID, lo
 	}
 	return maxVersion, nil
 }
+
+type AKSPublicIPPrefixSpecInput struct {
+	Cluster           *clusterv1.Cluster
+	KubernetesVersion string
+	WaitIntervals     []interface{}
+}
+
+func AKSPublicIPPrefixSpec(ctx context.Context, inputGetter func() AKSPublicIPPrefixSpecInput) {
```

Comment:
Great work on adding e2e testing here 💯 

I love how you added a reusable spec that reuses the existing test AKS cluster to add a new MachinePool with the specific feature you want to test, instead of creating a whole new cluster. 

Would you be willing to show/demo this concept at CAPZ office hours? I think we should consider adopting this concept more widely in our e2e tests.

Reply:
Thanks! I could definitely share this at office hours. @willie-yao has done a ton with this pattern to test VM extensions, so I could tag team with him if he wants.